scripts/install_build_tools.sh

```diff
@@ -4,24 +4,35 @@ set -e
 . /etc/lsb-release
 echo "Distribution code name : ${DISTRIB_CODENAME}"
 apt-get update
-if [ "$DISTRIB_CODENAME" = "bionic" ];then 
-    echo "Installing bionic build tools ..."
-    DEPENDENCIES=(
-        build-essential 
-        cmake 
-        googletest 
-        google-mock
-        gdb
-        git)
-else
-    echo "Installing other linux distribution (focal or jammy) build tools ..."
-    DEPENDENCIES=(
+apt-get -y install 
+DEPENDENCIES=(
+        wget
+        ca-certificates
         build-essential
-        cmake
-        build-essential
-        libgtest-dev
-        libgmock-dev
         gdb
         git)
-fi
-DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends --yes --quiet "${DEPENDENCIES[@]}"
\ No newline at end of file
+DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends --yes --quiet "${DEPENDENCIES[@]}"
+# Installing CMake 
+CMAKE_VERSION="3.27.7"
+echo "Installing CMake ${CMAKE_VERSION}"
+wget https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-x86_64.sh
+wget https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz
+chmod u+x cmake-${CMAKE_VERSION}-linux-x86_64.sh
+./cmake-${CMAKE_VERSION}-linux-x86_64.sh --skip-license --include-subdir --prefix=/opt
+rm cmake-${CMAKE_VERSION}-linux-x86_64.sh
+rm cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz
+echo "export PATH=/opt/cmake-${CMAKE_VERSION}-linux-x86_64/bin:$PATH" >> /root/.bashrc
```

Comment:
Why are we putting it in `root`'s `.bashrc` file? With this approach, the new CMake installation will only be found for `root`, other users will not find it. If you want to modify `PATH` for all users, it should be like this:

```shell
echo 'export PATH=/opt/cmake-'${VERSION}'-linux-'${ARCH}'/bin:$PATH' >> /etc/profile.d/cmake_path.sh
```

Reply:
I looked into Bash and its configuration files. When you run a Docker image, you get an interactive non-login shell, which will source `/etc/bashrc`. Try changing the line to 

```shell
echo "export PATH=/opt/cmake-${CMAKE_VERSION}-linux-x86_64/bin:$PATH" >> /etc/bashrc
```

Reply:
This didnt work either